### ISSUE TITLE
homectl: allow creating user accounts with just a fido2 token, without any password configured.

### ISSUE THREAD

**Issue by ksy63** [systemd/systemd#24206]
### systemd version the issue has been seen with

251

### Used distribution

Arch

### Linux kernel version used

5.18.15

### CPU architectures issue was seen on

x86_64

### Component

homectl

### Expected behaviour you didn't see

systemd-homed:
Using a fido2 device with hmac-secret support, a password should not be required for either login nor encryption.

### Unexpected behaviour you saw

During setup of a new user with a fido2 device, a user password is requested, therefore defeating the intended purpose of the security device to replace passwords as a single factor

### Steps to reproduce the problem

homectl create homeuser --storage=fscrypt --fido2-device=/dev/hidrawX

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
Initializing FIDO2 credential on security token.
👆 (Hint: This might require confirmation of user presence on security token.)
🔐 Please enter security token PIN: ********
Generating secret key on FIDO2 security token.
👆 In order to allow secret key generation, please confirm presence on security token.
🔐 Please enter new password for user homeuser: ******
🔐 Please enter new password for user homeuser (repeat): ******
👆 Please confirm presence on security token.
```

**Comment by poettering** [comment]
Yeah, we always ask for a password to be configured, since for man purposes you still kinda need one, since not everything is hooked up properly to PAM authentication. For example, SSH.

But do note that the password is not a 2nd factor. You can use the password *or* the authentication token. Not the combination.

That all said, it probably makes sense to have a --no-password switch or so, to be able to create accounts with no password at all, and only do FIDO or so.
